docs/guides/modding-guide.rst

```diff
@@ -0,0 +1,74 @@
+DFHack modding guide
+====================
+
+Preface
+-------
+
+Why not just use raw modding?
+
+It's much less hacky for many things, easier to maintain, and easier to extend. Of course if you're adding a new creature or whatever do use raw modding, but for things like adding syndromes when certain actions are performed, absolutely try out DFHack. Many things will require a mix of raw modding and DFHack.
```

Comment:
Could you characterize the "many things" it's less hacky for a little more? adding syndromes is one. What are the other major use cases? You mention "modify behaviours" below, but I think we can get more specific. Getting examples from other modders might help here.